@here/olp-sdk-core/lib/utils/multipartupload-internal/NodeFileData.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { BlobData, fs } from "@here/olp-sdk-core";
+
+/**
+ * @internal
+ * Implementation of reading bytes from file by filepath
+ * in Node.js.
+ */
+export class NodeFileData implements BlobData {
+    private fileHandle?: fs.promises.FileHandle;
+
+    constructor(private readonly filePath: string) {}
+
+    async readBytes(offset: number, count: number): Promise<ArrayBuffer> {
+        if (!this.fileHandle) {
+            this.fileHandle = await this.openFile();

```

Comment:
Instead of opening the file every time, which probably is very expensive, I would open it once in the constructor and keep the FileHandle for as long as the class instance is alive. If I read the documentation correctly node.js uses open() which returns a fd valid for the lifecycle of the process. So it is safe to keep it as long as you need it. But please double-check with the node.js FileHandle documentation.

Reply:
works as you described. We can't open in constructor, because it's async operation. We open once on the first call readBytes or size method and cache it.